### PR TITLE
ci: Only notify benchmarks failure via Slack on Nightly scheduled run

### PR DESCRIPTION
**Pull request by tungbq**
## Motivation/summary

Only notify benchmarks failure via Slack on Nightly scheduled run

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes: #11293

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -127,23 +127,23 @@ jobs:
         if: always()
         run: make destroy
 
-      - if: failure()
+        # Notify failure to Slack only on schedule (nightly run)
+      - if: failure() && github.event_name == 'schedule'
```
Comment (axw) [inline .github/workflows/benchmarks.yml:131]: I think we also need a similar `- if: github.event_name == 'schedule'` on the step below, for the success case.

**Review by axw — CHANGES_REQUESTED** [review]
Thanks for the PR @tungbq! Your change looks good, we just need to update the success step as well.

**Comment by axw** [comment]
@tungbq also, we would need you to sign the [CLA](https://www.elastic.co/contributor-agreement) before we can accept any contributions.

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (tungbq) [reply .github/workflows/benchmarks.yml:131]: It's done @axw

**Comment by tungbq** [comment]
> @tungbq also, we would need you to sign the [CLA](https://www.elastic.co/contributor-agreement) before we can accept any contributions.

Done! Thanks for your guidance @axw

**Review by axw — APPROVED** [review]
Thank you for your contribution @tungbq, much appreciated!

**Comment by axw** [comment]
@tungbq sorry, one more thing - some of your commits are not signed. Can you please fix that? You will probably need to squash your commits and force push.

**Comment by tungbq** [comment]
> @tungbq sorry, one more thing - some of your commits are not signed. Can you please fix that? You will probably need to squash your commits and force push.

Thanks for letting me know. I've updated this point as well @axw

**Comment by axw** [comment]
@tungbq the squashed commit is not signed either. Do you have commit signing set up locally? See https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

**Comment by tungbq** [comment]
> @tungbq the squashed commit is not signed either. Do you have commit signing set up locally? See https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

Sorry for the inconvenience. I've updated the commit. Can you check again?
image

**Comment by axw** [comment]
> Sorry for the inconvenience. I've updated the commit. Can you check again @axw?

No problem! Thank you again @tungbq!

**Pull request merged**